src/OpenTelemetry.Instrumentation.AspNetCore/Implementation/HttpInListener.cs

```diff
@@ -110,11 +110,29 @@ public override void OnStartActivity(Activity activity, object payload)
                 }
             }
 
-            ActivityInstrumentationHelper.SetActivitySourceProperty(activity, ActivitySource);
-            ActivityInstrumentationHelper.SetKindProperty(activity, ActivityKind.Server);
-
+            // enrich Activity from payload only if sampling decision
+            // is favorable.
             if (activity.IsAllDataRequested)
             {
+                try
+                {
+                    if (this.options.Filter?.Invoke(context) == false)
```

Comment:
if not exists, you can add a test validating that Filter is only called if sampling is favorable.. Same for Enrich.

Reply:
Added a unit test for testing both Filter and Enrich are only called when sampled.